content/blog/base-vs-magrittr-pipe/index.Rmd

```diff
@@ -0,0 +1,85 @@
+  author: Sigmund
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [other] 
+tags: [magrittr]
+---
+
+<!--
+TODO:
+* [x] Look over / edit the post's title in the yaml
+* [x] Edit (or delete) the description; note this appears in the Twitter card
+* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+**Note:** The following has been adapted from a section of the forthcoming second edition of [R for Data Science](https://r4ds.hadley.nz/) that had to be removed due to length limitations.
+
+## Pipes
+
+R 4.1.0 introduced a native pipe operator, `|>`.
+As described in the [R News](https://cran.r-project.org/doc/manuals/r-devel/NEWS.html):
+
+> R now provides a simple native forward pipe syntax `|>`. The simple form of the forward pipe inserts the left-hand side as the first argument in the right-hand side call. The pipe implementation as a syntax transformation was motivated by suggestions from Jim Hester and Lionel Henry.  
+
+The behaviour of the native pipe is by and large the same as that of the [`%>%`](https://magrittr.tidyverse.org/reference/pipe.html) pipe provided by the **magrittr** package. 
+Both operators (`|>` and `%>%`) let you "pipe" an object forward to a function or call expression, thereby allowing you to express a sequence of operations that transform an object.
+
+To learn more about the basic utility of pipes, see [The pipe](https://r4ds.hadley.nz/data-transform.html#the-pipe) section of R for Data Science.
+
+Luckily there's no need to commit entirely to one pipe or the other --- you can use the base pipe for the majority of cases where it's sufficient and use the magrittr pipe when you really need its special features.
+
+## `|>` vs. `%>%`
+
+While `|>` and `%>%` behave identically for simple cases, there are a few crucial differences.
+These are most likely to affect you if you're a long-term user of `%>%` who has taken advantage of some of the more advanced features.
+But they're still good to know about even if you've never used `%>%` because you're likely to encounter some of them when reading wild-caught code.
+
+-   By default, the pipe passes the object on its left-hand side to the first argument of the function on the right-hand side.
+    `%>%` allows you to change the placement with a `.` placeholder.
+    For example, `x %>% f(1)` is equivalent to `f(x, 1)` but `x %>% f(1, .)` is equivalent to `f(1, x)`.
+    R 4.2.0 added a `_` placeholder to the base pipe, with one additional restriction: the argument has to be named.
+    For example, `x |> f(1, y = _)` is equivalent to `f(1, y = x)`.
```

Comment:
R 4.3.0 adds an additional feature that allows you to use `_` in an extraction call like `_$x` or `_[["y"]]`.